llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -305,6 +308,10 @@ static bool hasAssertInFunctionCallGraph(llvm::Function *Func) {
     if (F != Func)
       FuncCallStack.push_back(F);
 
+    if (!HasIndirectlyCalledAssert &&
+        F->hasFnAttribute("referenced-indirectly"))
+      HasIndirectlyCalledAssert = true;
```

Comment:
Does it mean that even if `referenced-indirectly` function doesn't call the assert we will still mark it as one which call the assert?

Reply:
Thanks, that was a logical mistake - I fixed it in ffacf61

Reply:
Shouldn't we add a regression test for the scenario described by Alexey?

Reply:
@bader done in f62f0d0